docs: migrate 'metrics and statistics' (#5372)
canonical/snapcraft: docs/how-to/publish/get-snap-metrics.rst

.. _how-to-get-snap-metrics:

Get snap metrics
================

Snap stores collect installation statistics for snaps you authored, and you can retrieve
the collected data with Snapcraft.

For a complete list of available snap metrics, see :ref:`reference-metrics`.

Prerequisites
-------------

> As snap metrics are confidential, and only a snap's author can access them.

Reviewer: ```suggestion
As snap metrics are confidential, only a snap's author can access them.
```

> As snap metrics are confidential, and only a snap's author can access them.

Reviewer: ```suggestion
As snap metrics are confidential, only a snap's author can access them.
```

> Before you begin, log in to your Snapcraft account.

Reviewer: ```suggestion
Before you begin, log in to your Snapcraft account by running:

.. code-block:: bash

    snapcraft login
    
```
I believe we've mentioned this before, but this comes up often enough that it would be nice to have a page for account creation and login.

Author: This page is deep enough into the tasks that we won't need to repeat it. I've left a comment for a placeholder link: 1bf114a

The returned data then takes the form of a JSON object:

.. collapse:: JSON object for a metric

    .. code-block:: json

      {
        "buckets": [
          "2025-03-03",
          "2025-03-04",
          "2025-03-05",
          "2025-03-06",
          "2025-03-07",
          "2025-03-08"
        ],
        "metric_name": "installed_base_by_version",
        "series": [
          {
            "name": "v0.27.0",
            "values": [
              21,
              20,
              22,
              16,
              null,
              null
            ]
          },
          {
            "name": "v0.28.2",
            "values": [
              null,
              null,
              null,
              3,
              19,
              20
            ]
          }
        ],
        "snap_id": "q9HYHk05OMrPAvzSb1q6AXkmZR6rkDgx",
        "status": "OK"
      }

Troubleshoot access to metric data
----------------------------------

The `Snap Store metrics API
<https://dashboard.snapcraft.io/docs/reference/v1/snap.html#fetch-metrics-for-snaps>`_,
called by the ``snapcraft metrics`` command, requires your account to have the
package_metrics permission.

> If this is your first time querying a snap metric after you registered your Snapcraft

Reviewer: ```suggestion
If this is your first time querying a snap metric after registering your Snapcraft
```

Author: Used a different conjugation: 1bf114a
